src/main/java/com/google/api/gax/core/FlowController.java

```diff
@@ -87,23 +96,43 @@ public String toString() {
     }
   }
 
+  public enum LimitExceededBehavior {
+    ThrowException,
+    Block,
+  }
+
   @Nullable private final Semaphore outstandingElementCount;
   @Nullable private final Semaphore outstandingByteCount;
+  private final boolean flowControlEnabled;
```

Comment:
A part of me thinks this shouldn't be a flag. If `settings.getIsEnabled()` is false, we can just set the counts and semaphores to null. I feel like making separate flag just makes us check multiple flags instead of one.